fw/apm.c

```diff
@@ -1360,16 +1374,19 @@ tfw_apm_hm_srv_alive(int status, TfwStr *body, struct sk_buff *skb_head,
 	 * Special case for 'auto' monitor: generate crc32
 	 * from body of first response and store it into monitor.
 	 */
-	if (!hm->crc32 && hm->auto_crc) {
-		hm->crc32 = __tfw_apm_crc32_calc(&it, &chunk, skb_head, body);
+	if (hm->crc32 == 0 && hm->auto_crc) {
```

Comment:
Does the original variant leads to some sort of UB? Just asking.

Reply:
`skb_head` is unused argument. And it presence was a motivation to review and change `tfw_apm_hm_srv_alive()`
 signature. 
 The other is just a pinch of readability.

Reply:
Oh, I meant `!hm->crc32` as the original variant versus `hm->crc32 == 0`. Just didn't see the sense in changing it unless there's UB that I don't know of. :-) Usually these kinds of micro-changes are not encouraged and the original variant is a typical idiomatic C language construction.

Reply:
Idiomatic :)
Classic idiomas are changing.
https://github.com/tempesta-tech/tempesta/issues/1695#issuecomment-2014672322
Agree, it is not this case, but I've already changed these piece of code and direct comparation  show that is numeric value, not pointer or boolean.

Reply:
There is no sense to make these changes. [#1695 (comment)](https://github.com/tempesta-tech/tempesta/issues/1695#issuecomment-2014672322) is about address comparison, it's not related to primitive types.

Even for pointers we still can use `(!p)`, `-Waddress` intended to catch such cases: `NULL + 1 == NULL`

Reply:
OK